compiler/ghc/default.nix

```diff
@@ -354,38 +390,77 @@ stdenv.mkDerivation (rec {
+      + ''
+        mkdir -p $generated/libraries/ghc-boot/dist-install/build/GHC/Platform
+        if [[ -f _build/stage1/libraries/ghc-boot/build/GHC/Version.hss ]]; then
+          cp _build/stage1/libraries/ghc-boot/build/GHC/Version.hs $generated/libraries/ghc-boot/dist-install/build/GHC/Version.hs
+        fi
+        if [[ -f _build/stage1/libraries/ghc-boot/build/GHC/Platform/Host.hs ]]; then
+          cp _build/stage1/libraries/ghc-boot/build/GHC/Platform/Host.hs $generated/libraries/ghc-boot/dist-install/build/GHC/Platform/Host.hs
+        fi
+      ''
+      # Convert ${pkgroot} relative paths to /nix/store paths
+      + ''
+        for f in "$out/${rootDir}lib/package.conf.d/"*.conf; do
+          sed -i -e "s|\''${pkgroot}/../lib/|$out/${rootDir}lib/|" \
+                 -e "s|\''${pkgroot}/../share/|$out/${rootDir}share/|" \
+                 -e "s|\''${pkgroot}/../../../share/doc/|$doc/share/doc/|" $f
+        done
+      ''
+    else
+      ''
+        # Save generated files for needed when building ghc and ghcjs
+        mkdir -p $generated/includes/dist-derivedconstants/header
+        cp includes/dist-derivedconstants/header/GHCConstantsHaskell*.hs \
+           $generated/includes/dist-derivedconstants/header
+        if [[ -f includes/ghcplatform.h ]]; then
+          cp includes/ghcplatform.h $generated/includes
+        elif [[ -f includes/dist-install/build/ghcplatform.h ]]; then
+          cp includes/dist-install/build/ghcplatform.h $generated/includes
+        fi
+        mkdir -p $generated/compiler/stage2/build
+        cp compiler/stage2/build/Config.hs $generated/compiler/stage2/build || true
+        if [[ -f compiler/stage2/build/GHC/Platform/Constants.hs ]]; then
+          mkdir -p $generated/compiler/stage2/build/GHC/Platform
+          cp compiler/stage2/build/GHC/Platform/Constants.hs $generated/compiler/stage2/build/GHC/Platform
+        fi
+        if [[ -f compiler/stage2/build/GHC/Settings/Config.hs ]]; then
+          mkdir -p $generated/compiler/stage2/build/GHC/Settings
+          cp compiler/stage2/build/GHC/Settings/Config.hs $generated/compiler/stage2/build/GHC/Settings
+        fi
+        cp compiler/stage2/build/*.hs-incl $generated/compiler/stage2/build || true
+        mkdir -p $generated/rts/build
+        cp rts/build/config.hs-incl $generated/rts/build || true
+
+        # Save generated files for needed when building ghc-boot
+        mkdir -p $generated/libraries/ghc-boot/dist-install/build/GHC/Platform
+        if [[ -f libraries/ghc-boot/dist-install/build/GHC/Version.hs ]]; then
+          cp libraries/ghc-boot/dist-install/build/GHC/Version.hs $generated/libraries/ghc-boot/dist-install/build/GHC/Version.hs
+        fi
+        if [[ -f libraries/ghc-boot/dist-install/build/GHC/Platform/Host.hs ]]; then
+          cp libraries/ghc-boot/dist-install/build/GHC/Platform/Host.hs $generated/libraries/ghc-boot/dist-install/build/GHC/Platform/Host.hs
+        fi
+    ''}
```

Comment:
For this whole section, can we add a comment, what the purpose of saving the files for is? I'm a bit lost as to why we do/need this.

Reply:
They are only needed for building ghcjs as a stand alone package.  I have added a comment.

Reply:
Oh also for the reinstallable lib ghc package in overlays/ghc-packages.nix.  I have updated the comment.